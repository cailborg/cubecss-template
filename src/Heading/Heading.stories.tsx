import { Story, Meta } from "@storybook/react/types-6-0";
import { Heading, HeadingProps } from "./Heading";
import { Text } from "../Text/Text"

export default {
  title: "Components/Heading",
  component: Heading,
  argTypes: {
    children: { control: "text" },
  },
} as Meta;

export const Standard: Story<HeadingProps> = (args) => <Heading {...args}>{args.children}</Heading>;
Standard.args = { children: 'Example Text', variant:"jumbo", tagName:"h1", alignment: "left" };

export const withText: Story<HeadingProps> = (args) => <><Heading {...args}>{args.children}</Heading><Text variant="body-1" tagName="p">Here is a paragrah</Text></>;
withText.args = { children: 'Example Text', variant:"jumbo", tagName:"h1", alignment: "left" };