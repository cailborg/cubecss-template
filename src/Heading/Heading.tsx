import React from "react";
import styled from "@emotion/styled";

export interface HeadingProps {
    variant?: "jumbo" | "xxxl" | "xxl" | "xl" | "lg" | "md" | "sm" | "xs";
    children: React.ReactNode;
    tagName?: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
    alignment?: "left" | "right" | "center";
    id?: string;
    className?: string;
}
export const Heading: React.FC<HeadingProps> = ({
    variant,
    children,
    tagName = "h1",
    alignment,
    id,
    className,
}) => {
    const HeadingTag = styled.h1`
        text-align: ${alignment ?? "inherit"};
        font-family: var(--lp-fonts-heading-${variant});
        font-size: var(--lp-font-size-heading-${variant});
        line-height: var(--lp-line-height-heading-${variant});
        letter-spacing: var(--lp-letter-spacing-heading-${variant});
        font-weight: var(--lp-font-weight-heading-${variant});
        & + p {
            margin-top: var(--lp-paragraph-spacing-heading-${variant});
        }
    `;
    return (
        <HeadingTag className={className} as={`${tagName}`} id={id}>
            {children}
        </HeadingTag>
    );
};
