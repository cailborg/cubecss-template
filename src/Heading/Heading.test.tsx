import React from "react";
import { Heading } from "./Heading";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<Heading>Hello</Heading>).toJSON();
  expect(tree).toMatchSnapshot();
});
