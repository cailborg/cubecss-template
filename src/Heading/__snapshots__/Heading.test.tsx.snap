// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<h1
  className="css-q913o6-Heading"
>
  Hello
</h1>
`;
